spec/features/process_a_request_accept_with_contact_list_spec.rb

```diff
@@ -94,6 +96,27 @@
       expect(vst).to be_booked
       expect(vst.nomis_id).to be_nil
     end
+
+    scenario 'available date the prisoner has a closed restriction', vcr: { cassette_name: 'closed_restriction' } do
+      visit prison_visit_path(vst, locale: 'en')
+
+      expect(page).to have_css('.choose-date .tag--error', text: 'Closed visit restriction')
+
+      choose_date
+
+      within "#visitor_#{visitor.id}" do
+        select 'IRMA ITSU - 03/04/1975', from: 'Match to contact list'
+      end
+
+      expect(page).to have_css('.panel', text: "This is a closed visit. Book this visit into NOMIS, then enter the reference number")
+
+      click_button 'Process'
+      expect(page).to have_css('.notification', text: 'Thank you for processing the visit')
+
+      vst.reload
+      expect(vst).to be_booked
```

Comment:
👍 on relying less on database access to do the checks. I think this is a leftover from when the app didn't have a staff UI other than the processing page.

There are some things that we check that we can only know from the database such as the noms visit id.